### DISCUSSION TITLE
convert username to snowflake/ID

### DISCUSSION THREAD

**Discussion by CSutter5** [Rapptz/discord.py#6378] [Q&A]
How can I convert a username to a snowflake/ID

**Comment by bandwiches** [comment]
Are you trying to turn a username string into a `User` object?

**Reply by CSutter5** [reply]
yes

**Reply by bandwiches** [reply]
Is the username associated with a guild member or can the username just be supplied at the end of a command like: `!bot lookup username#1234` to fetch a specified non-associated user?

**Reply by CSutter5** [reply]
it is not associated with a guild member, and can be supplied in a command

**Reply by bandwiches** [reply]
You can use [discord.Object](https://discordpy.readthedocs.io/en/latest/api.html#discord.Object) to create an object that's based on a `snowflake`, but you have to pass the `id`.

[Client Objects](https://discordpy.readthedocs.io/en/latest/api.html#client) provide a way to fetch users, but ultimately you're still stuck without the `id`.
